app/controllers/settings_controller.rb

```diff
@@ -256,6 +256,30 @@ def twitter_disconnect
     return redirect_to "/settings"
   end
 
+  def freenode_callback
+    reader = FreenodeTaxonomyReader.new
+    taxonomy = reader.for_user(params[:user][:freenode_username])
+
+    if taxonomy['LOBSTERS'] == @user.username
+      @user.freenode_username = params[:user][:freenode_username]
+      @user.save!
+      flash[:success] = "Your account has been linked to freenode user #{@user.freenode_username}."
+    elsif taxonomy['LOBSTERS']
+      flash[:error] = "Expected #{@user.username} in the taxonomy, not #{taxonomy['LOBSTERS']}."
+    else
+      flash[:error] = "The LOBSTERS key was not in #{params[:user][:freenode_username]}'s taxonomy."
+    end
+
+    return redirect_to "/settings"
+  end
+
+  def freenode_disconnect
+    @user.freenode_username = nil
+    @user.save!
+    flash[:success] = "Your freenode association has been removed."
+    return redirect_to "/settings"
```

Comment:
Let's save them a click and link to their profile, because they're going to immediately go click on their profile to see it gone. :D

Reply:
Should we update all of the service integrations to do that too?

Reply:
After this is merged we can make an issue for it, no need to do the others now.